sdk/core/Azure.Core.TestFramework/src/RecordedTestSanitizer.cs

```diff
@@ -76,9 +91,14 @@ public virtual string SanitizeTextBody(string contentType, string body)
                     foreach (JToken token in jsonO.SelectTokens(jsonPath))
                     {
                         token.Replace(sanitizer(token));
+                        modified = true;
                     }
                 }
-                return JsonConvert.SerializeObject(jsonO, SerializerSettings);
+
+                if (modified || LegacyConvertJsonDateTokens)
```

Comment:
nit: consider using braces here since we do that elsewhere in this same method/file for 1 line statements.